### DISCUSSION TITLE
Caption auto size without new lines

### DISCUSSION THREAD

**Discussion by M1chael** [ImageMagick/ImageMagick#2040] [Help] [answered]
Hi! 👋

I need a caption to be auto-sized to the box with particular dimensions. IM can do it with [this nice trick](http://www.imagemagick.org/Usage/text/#caption_bestfit). But I need to prevent it from adding new lines to the text.

A poem is a good example of text formatting I need. Suppose we have some text with newlines:

> Twinkle, twinkle, little star,
> How I wonder what you are.
> Up above the world so high,
> Like a diamond in the sky.

So I want IM to write it in the box with a particular size and not adding new lines to it like that:

> Twinkle, twinkle, little star,
> How I wonder what you 
> are.
> Up above the world so 
> high,
> Like a diamond in the 
> sky.

Is there any way to do it with IM?

**Comment by snibgo** [comment, marked as answer]
`magick label:"Twinkle, twinkle, little star,\nHow I wonder what you are.\nUp above the world so high,\nLike a diamond in the sky." out.png`
